src/brsTypes/components/RoSGNode.ts

```diff
@@ -325,6 +325,43 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
         return false;
     }
 
+    /**
+     * Calls the function specified on this node.
+     */
+    private callfunc = new Callable("callfunc", {
+        signature: {
+            args: [
+                new StdlibArgument("functionname", ValueKind.String),
+                new StdlibArgument("functionargs", ValueKind.Dynamic, BrsInvalid.Instance),
+            ],
+            returns: ValueKind.Dynamic,
+        },
+        impl: (interpreter: Interpreter, functionname: BrsString, functionargs: BrsType) => {
+            // We need to search the callee's environment for this function rather than the caller's.
+            let componentDef = interpreter.environment.nodeDefMap.get(this.nodeSubtype);
+
+            // Only allow public functions (defined in the interface) to be called.
+            if (componentDef && functionname.value in componentDef.functions) {
+                return interpreter.inSubEnv(subInterpreter => {
+                    let functionToCall = subInterpreter.getCallableFunction(functionname.value);
+
+                    // Determine whether the function should get arguments are not.
+                    if (functionToCall.getFirstSatisfiedSignature([functionargs])) {
+                        return functionToCall.call(subInterpreter, functionargs);
+                    } else {
+                        return functionToCall.call(subInterpreter);
+                    }
+                }, componentDef.environment);
+            } else {
+                console.error(
+                    `Warning calling function in ${this.nodeSubtype}: no function interface specified for ${functionname}`
+                );
```

Comment:
Pro tip for next time: if you use `interpreter.stderr.write` instead of `console.error`, you won't have to mock anything in your tests!  Just `outputStreams.stderr` the same way we handle `stdout` in the e2e tests 😄 

I'll update this line before I merge though!

Reply:
done!